streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamTaskTest.java

```diff
@@ -2187,7 +2187,7 @@ public void shouldUnregisterMetricsInCloseCleanAndRecycleState() {
 
         task.suspend();
         assertThat(getTaskMetrics(), not(empty()));
-        task.recycleAndConvert();
+        task.prepareRecycle();
```

Comment:
The following comment is not related to this PR, but I want to mention it nevertheless for awareness. In `StreamTaskTest` and `StandbyTaskTest` we have some gaps in testing `prepareRecycle()`. Most notably, we do not verify the state change to CLOSE and the recording of the `closeTaskSensor`. However, verification of this behavior is important for refactorings.

Reply:
ack.